src/InputGroup/index.js

```diff
@@ -151,126 +159,134 @@ StyledInputGroup.defaultProps = {
   theme: defaultTheme,
 };
 
-class InputGroup extends React.PureComponent<Props, State> {
-  state = {
-    active: false,
-    filled: false,
-  };
-
-  inputID: string = randomID("inputGroupID");
-
-  componentDidMount() {
-    this.isFilled();
-  }
-
-  componentDidUpdate(prevProps: Props) {
-    if (this.props.children !== prevProps.children) {
-      this.isFilled();
-    }
-  }
-
-  isFilled = () =>
-    this.setState({
-      filled: !React.Children.map(
-        this.props.children,
-        child => child.props.value !== undefined && child.props.value !== "",
-      ).includes(false),
-    });
-
-  handleFocus = (ev: SyntheticInputEvent<HTMLInputElement>) => {
-    const { onFocus } = this.props;
-
-    this.setState({ active: true });
+const InputGroup = ({
+  children,
+  label,
+  flex = "0 1 auto",
+  size = SIZE_OPTIONS.NORMAL,
+  help,
+  error,
+  dataTest,
+  spaceAfter,
+  onFocus,
+  onBlur,
+  onChange,
+}: Props) => {
+  const [active, setActive] = useState(false);
+  const [filled, setFilled] = useState(false);
+  const [tooltipShown, setTooltipShown] = useState(false);
+  const [tooltipShownHover, setTooltipShownHover] = useState(false);
+  const labelRef = useRef(null);
+  const iconRef = useRef(null);
+  const inputID: string = useMemo(() => randomID("inputFieldID"), []);
+
+  const isFilled = useCallback(() => {
+    return () => {
```

Comment:
Unnecessary return of function, e.g. `isFilled()` won't call the `setFilled`. IMHO, it could be done also with just `useMemo`.